xdsl/dialects/math.py

```diff
@@ -529,12 +462,18 @@ class FPowIOp(IRDLOperation):
     """
 
     name = "math.fpowi"
+
+    T1: ClassVar = VarConstraint("T1", floatingPointLike)
+    T2: ClassVar = VarConstraint("T2", signlessIntegerLike)
+
     fastmath = opt_prop_def(FastMathFlagsAttr)
```

Comment:
A lot of the other fastmath flags in mlir are default valued props instead of optional props. Is the math dialect different?

Reply:
What do you mean by "a lot"? Our arith and llvm dialect uses of these differ, but the net result is the same as absence means `none`.

I'll check MLIR later and amend in a separate PR just for the flags as required.
I also wanted to check the relevant interface, so another incentive to do so.